Make failed

# Compiled in Nobara 42 (based on Fedora 42)

```bash
░▒▓ 🎩  …/hikikom0ri/Descargas/mdbtools-1.0.1   ▓▒░ ◦•··························································································•◦ ░▒▓   21ms ▓▒░
··• ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether sleep supports fractional seconds... yes
checking filesystem timestamp resolution... 0.01
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking xargs -n works... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking for thread local storage (TLS) class... thread_local
checking whether make sets $(MAKE)... (cached) yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for flex... flex
checking for lex output file root... lex.yy
checking for lex library... none needed
checking whether yytext is a pointer... yes
checking for bison... bison -y
checking for fcntl.h... yes
checking for limits.h... yes
checking for unistd.h... (cached) yes
checking for xlocale.h... no
checking for DBLCMapStringW in -lmswstr... no
checking for gcc options needed to detect all undeclared functions... none needed
checking whether program_invocation_short_name is declared... yes
checking for a readline compatible library... yes
checking for readline.h... no
checking for readline/readline.h... no
checking whether readline supports history... yes
checking for history.h... no
checking for readline/history.h... no
checking for strptime... yes
checking for fmemopen... yes
checking for gmtime_r... yes
checking for reallocf... no
checking for wcstombs_l... no
checking for mbstowcs_l... no
checking for vasprintf... yes
checking for vasnprintf... no
checking whether the target supports symbol aliases... yes
checking for gcc option to enable large file support... none needed
checking Are we using flex ... yes 
checking for ld... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking for shared library run path origin... done
checking 32-bit host C ABI... no
checking how to run the C preprocessor... gcc -E
checking for egrep -e... (cached) /usr/bin/grep -E
checking for ELF binary format... yes
checking for the common suffixes of directories in the library search path... lib,lib,lib64
checking for iconv... yes
checking for working iconv... yes
checking whether iconv is compatible with its POSIX signature... yes
checking whether /usr/bin/ld -m elf_x86_64 accepts --as-needed... yes
checking pkg-config m4 macros... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for glib-2.0 >= 2.68... no
checking for bash-completion >= 2.0... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating src/Makefile
config.status: creating include/mdbtools.h
config.status: creating libmdb.pc
config.status: creating libmdbsql.pc
config.status: creating src/util/Makefile
config.status: creating src/util/bash-completion/Makefile
config.status: creating src/extras/Makefile
config.status: creating Makefile
config.status: creating include/Makefile
config.status: creating src/libmdb/Makefile
config.status: creating src/sql/Makefile
config.status: creating src/odbc/Makefile
config.status: creating src/fuzz/Makefile
config.status: creating doc/Makefile
config.status: creating include/mdbver.h
config.status: executing depfiles commands
config.status: executing libtool commands
configure: 
configure: MDB Tools 1.0.1 - Configuration summary
configure: 
configure:    SQL             : enabled
configure:    ODBC            : disabled
configure:    GLib            : disabled
configure:    iconv           : enabled
configure:    man pages       : enabled
configure:    bash-completion : enabled
configure: 
```
```bash
░▒▓ 🎩  …/hikikom0ri/Descargas/mdbtools-1.0.1   ▓▒░ ◦•·······················································································•◦ ░▒▓   6s275ms ▓▒░
··• make
Making all in src
make[1]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src'
Making all in libmdb
make[2]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/libmdb'
  CC       catalog.lo
  CC       file.lo
  CC       table.lo
  CC       data.lo
  CC       backend.lo
  CC       money.lo
  CC       sargs.lo
  CC       index.lo
  CC       like.lo
  CC       write.lo
  CC       stats.lo
  CC       map.lo
  CC       props.lo
  CC       worktable.lo
  CC       options.lo
  CC       iconv.lo
  CC       version.lo
  CC       rc4.lo
  CCLD     libmdb.la
make[2]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/libmdb'
Making all in extras
make[2]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/extras'
  CCLD     mdb-hexdump
make[2]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/extras'
Making all in sql
make[2]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/sql'
make  all-am
make[3]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/sql'
  CC       mdbsql.lo
  CC       parser.lo
  CC       lexer.lo
  CCLD     libmdbsql.la
make[3]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/sql'
make[2]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/sql'
Making all in util
make[2]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util'
Making all in bash-completion
make[3]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util/bash-completion'
make[3]: No se hace nada para 'all'.
make[3]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util/bash-completion'
make[3]: se entra en el directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util'
  CC       mdb-export.o
  CCLD     mdb-export
  CC       mdb-array.o
  CCLD     mdb-array
  CC       mdb-schema.o
  CCLD     mdb-schema
  CC       mdb-tables.o
  CCLD     mdb-tables
  CCLD     mdb-parsecsv
  CC       mdb-header.o
  CCLD     mdb-header
  CC       mdb-ver.o
  CCLD     mdb-ver
  CC       mdb-prop.o
  CCLD     mdb-prop
  CC       mdb-count.o
  CCLD     mdb-count
  CC       mdb-queries.o
  CCLD     mdb-queries
  CC       mdb-json.o
  CCLD     mdb-json
  CC       mdb-sql.o
mdb-sql.c: En la función ‘main’:
mdb-sql.c:379:17: error: too many arguments to function ‘read_history’; expected 0, have 1
  379 |                 read_history(histpath);
      |                 ^~~~~~~~~~~~ ~~~~~~~~
mdb-sql.c:42:12: nota: se declara aquí
   42 | extern int read_history ();
      |            ^~~~~~~~~~~~
mdb-sql.c:441:35: error: too many arguments to function ‘readline’; expected 0, have 1
  441 |                         char *l = readline(prompt);
      |                                   ^~~~~~~~ ~~~~~~
mdb-sql.c:28:14: nota: se declara aquí
   28 | extern char *readline ();
      |              ^~~~~~~~
mdb-sql.c:487:33: error: too many arguments to function ‘add_history’; expected 0, have 1
  487 |                                 add_history(s);
      |                                 ^~~~~~~~~~~ ~
mdb-sql.c:40:13: nota: se declara aquí
   40 | extern void add_history ();
      |             ^~~~~~~~~~~
mdb-sql.c:511:17: error: too many arguments to function ‘write_history’; expected 0, have 1
  511 |                 write_history(histpath);
      |                 ^~~~~~~~~~~~~ ~~~~~~~~
mdb-sql.c:41:12: nota: se declara aquí
   41 | extern int write_history ();
      |            ^~~~~~~~~~~~~
make[3]: *** [Makefile:698: mdb-sql.o] Error 1
make[3]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util'
make[2]: *** [Makefile:734: all-recursive] Error 1
make[2]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src/util'
make[1]: *** [Makefile:392: all-recursive] Error 1
make[1]: se sale del directorio '/home/hikikom0ri/Descargas/mdbtools-1.0.1/src'
make: *** [Makefile:492: all-recursive] Error 1
```

--- a/src/util/mdb-sql.c
+++ b/src/util/mdb-sql.c
@@ -25,7 +25,7 @@
 #    include <readline.h>
 #  else
 /* no readline.h */
-extern char *readline ();
+extern char *readline (const char *);
 #  endif
 char *cmdline = NULL;
 #endif /* HAVE_LIBREADLINE */
@@ -37,10 +37,10 @@ char *cmdline = NULL;
 #    include <history.h>
 #  else
 /* no history.h */
-extern void add_history ();
-extern int write_history ();
-extern int read_history ();
-extern void clear_history ();
+extern void add_history (const char *);
+extern int write_history (const char *);
+extern int read_history (const char *);
+extern void clear_history (void);
 #  endif
 #endif /* HAVE_READLINE_HISTORY */
 

